src/core/overflowsafe_type.hpp

```diff
@@ -39,25 +39,41 @@ class OverflowSafeInt
 	/**
 	 * Safe implementation of addition.
 	 * @param other the amount to add
-	 * @note when the addition would yield more than T_MAX (or less than T_MIN),
-	 *       it will be T_MAX (respectively T_MIN).
+	 * @note when the addition would yield more than T_MAX, it will be T_MAX.
```

Comment:
this commit message is a bit weird. `Fix: OverflowSafeInt behaviour with underflows `. and there really isn't much more context what this commit is trying to address, except "something" with underflows :) Maybe spend a few words on it in the commit message, or reword it to make it a bit more clear what was happening here?

The PR also says "potential" undefined behaviour .. which always makes me wonder: is it really potential? :D

Mainly I am trying to understand what this commit fixes, and it is really difficult to see :P Math hard .. :D

Reply:
"potential" comes from the fact that you only get undefined behaviour if you over/underflow. Not at all in 'normal' usage

Reply:
Suggestion:

```
Fix: OverflowSafeInt's underflow protection doesn't work in all cases

For example: INT32_MIN - 1 != INT32_MIN, while it would be expected
to be INT32_MIN.
```